### DISCUSSION TITLE
Problem with OMEdit depending on package layout

### DISCUSSION THREAD

**Discussion by franko** [OpenModelica/OpenModelica#12209] [General]
Hi all,

I am using OpenModelica v1.22.3 (64-bit) on Windows with OMEdit.

I recently changed the package layout from a .mo file to a directory with a package.mo. The goal was to include resources for documentation and lookup tables.

This change works well, except I'm now encountering issues with OMEdit. When I duplicate some models from the package, the GUI is no longer able to display and operate correctly with the package. In the tree representation on the right, some elements have disappeared. Additionally, I receive the following error in the console:

```
[3] 08:25:13 Syntax Error
[Motor.Motor: 246:0-246:0]: [No viable alternative near token: <EOF>](omeditmessagesbrowser:///Motor.Motor?lineNumber=246)
```

At this point when I click on any of the remaining visible elements, the GUI fails to display anything, and I encounter new errors such as:

```
[10] 08:25:36 Translation Error
Class Motor.VoltageSource.PatternEM1324 not found in scope <top>.
```

This issue does not occur when I lay out the package as a single .mo file, without utilizing a directory. In this case I can duplicate any model from the package and the issue never occurs.

I'm pretty stumped here and could really use your expertise. Has anyone else faced similar issues or have any tips on how to navigate this problem? I'm all ears for any suggestions or help you might offer. Thanks in advance for your support!

**Comment by casella** [comment]
@franko we recently improved the Duplicate feature. Are you using the latests nightly build?

**Reply by franko** [reply]
Ok, I tested right now with the nightly build and I still have the same problem. When I duplicate one of the model within the package a still get the error:

```
[4] 13:37:40 Syntax Error
[Motor.System: 298:0-298:0]: [No viable alternative near token: <EOF>](omeditmessagesbrowser:///Motor.System?lineNumber=298)

[5] 13:37:40 Syntax Error
[Motor.Utilities: 1:6-1:12]: [Parser error: Unexpected token near: import (IMPORT)](omeditmessagesbrowser:///Motor.Utilities?lineNumber=1)
```

If you have some suggestions to how to debug the problem that would be welcome. Currently I have no idea abut what exactly triggers the problem.

The line 298 in the error has nothing special, here the fragment:

```
    end HorIC;
    annotation(
      Icon(graphics = {Line(points = {{-80, -16}, {-4, -16}, {-4, 28}, {38, 28}, {38, -16}, {66, -16}}, color = {255, 0, 255})}));
  end Driver;

  package Circuit
    extends Modelica.Icons.Package;

    model PontHIdeal
```

it is the line where "package Circuit" is located.

I thought I may split the package in smaller subpackages defined in separated files but I don't know.

**Reply by casella** [reply]
@franko it would be great if you could share the code that causes you trouble, so we can replicate it.

**Reply by franko** [reply]
Well, unfortunately that would be problematic because the package includes IP elements I cannot share publicly.

**Reply by casella** [reply]
You can try to remove/rename/obfuscate stuff until it's pretty useless, as long as you still get this issue. Or, you can send it to us via email, we won't share it.

Without a concrete example of code to debug, it is hard to figure out what is going wrong. @perost, any clue?

**Reply by perost** [reply]
The error messages look weird since the filenames are all wrong, but maybe that's edited? But otherwise it looks like something's gone wrong with the file since EOF = end of file. That's not really something the duplicate feature itself can cause since it just generates Absyn structures, so I don't really have any clue on what might be going on.

**Comment by franko** [comment]
Just for information, for people who made comments or those reading this thread:

I was able to solve the problem using a partitioning approach: I removed selectively a model or a subpackage from the package until I found the one that caused the problem with the OMEdit GUI. At the end I removed a small subpackage that I reconstructed later and the problem disappeared.

**Reply by casella** [reply]
Divide & Conquer, a.k.a. bisection. Always works 😃
